internal/constraints/cache.go

```diff
@@ -44,21 +46,45 @@ func (c *Cache) Build(
 	env *cel.Env,
 	fieldDesc protoreflect.FieldDescriptor,
 	fieldConstraints *validate.FieldConstraints,
+	extensionTypeResolver protoregistry.ExtensionTypeResolver,
+	allowUnknownFields bool,
 	forItems bool,
 ) (set expression.ProgramSet, err error) {
-	constraints, done, err := c.resolveConstraints(fieldDesc, fieldConstraints, forItems)
+	constraints, done, err := c.resolveConstraints(
+		fieldDesc,
+		fieldConstraints,
+		forItems,
+	)
 	if done {
 		return nil, err
 	}
 
+	if err = reparseUnrecognized(extensionTypeResolver, constraints); err != nil {
+		return nil, errors.NewCompilationErrorf("error reparsing message: %w", err)
+	}
+	if !allowUnknownFields && len(constraints.GetUnknown()) > 0 {
+		return nil, errors.NewCompilationErrorf("unknown constraints in %s", constraints.Descriptor().FullName())
```

Comment:
Do we also want to suggest the `WithAllowUnknownFields` option in the error message?

Reply:
I added a note in d021bd5a7eb6b3ce628651f891cfbccc7745bc86. To keep it succinct I only mention `WithExtensionTypeResolver` _but_ I add a note to the `WithExtensionTypeResolver` option that points to `WithAllowUnknownFields`.

Reply:
Nice, thanks